modules/core/src/main/java/org/apache/ignite/internal/client/thin/ReliableChannel.java

```diff
@@ -70,8 +70,11 @@ final class ReliableChannel implements AutoCloseable {
     /** Limit of attempts to execute each service. */
     private volatile int attemptsLimit;
 
-    /** Index of the current channel. */
-    private volatile int curChIdx = -1;
+    /** Index of the current channel (for round-robin balancing). */
+    private final AtomicInteger curChIdx = new AtomicInteger();
+
+    /** Index of the default channel. */
+    private volatile int defaultChIdx = -1;
```

Comment:
If this variable is changed every time a request is made, the name `defaultChIdx` is not appropriate.